sys/include/net/dns_mock.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2022 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_dns_mock DNS defines
+ * @ingroup     net
+ * @brief       Generic DNS mock values
+ * @{
+ *
+ * @file
+ * @brief   Generic DNS mock values
+ *
+ * @author  Hendrik van Essen <hendrik.ve@fu-berlin.de>
+ */
+#ifndef NET_DNS_MOCK_H
+#define NET_DNS_MOCK_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "net/ipv4/addr.h"
+#include "net/ipv6/addr.h"
+
+/**
+ * @brief Hostname used to query when using sock_dns_mock
+ */
+#define SOCK_DNS_MOCK_EXAMPLE_COM_HOSTNAME "example.com"
+
+/**
+ * @brief IPv4 string related to the hostname when using sock_dns_mock
+ *
+ * @see @ref SOCK_DNS_MOCK_EXAMPLE_COM_HOSTNAME
+ */
+#define SOCK_DNS_MOCK_EXAMPLE_COM_IPV4_STRING "93.184.216.34"
+
+/**
+ * @brief IPv6 string related to the hostname when using sock_dns_mock
+ *
+ * @see @ref SOCK_DNS_MOCK_EXAMPLE_COM_HOSTNAME
+ */
+#define SOCK_DNS_MOCK_EXAMPLE_COM_IPV6_STRING "2606:2800:220:1:248:1893:25c8:1946"
```

Comment:
What are those used for?

Reply:
At one point I thought about parsing the `ipv4_addr_t` into a string and compare it on a string basis, but I decided that this is an unnecessary step and that comparing it by `ipv4_addr_equal` is sufficient already. I kept them because in my opinion a define does not hurt and may be used later, but we can also remove these and simply mention the IP address in the doc of `sock_dns_mock_example_com_addr_ipv4` and `sock_dns_mock_example_com_addr_ipv6`.

Reply:
Let’s remove it - I can easily see this getting out of sync with the actual address in the future.

Reply:
Removed and adapted documentation